exercises/card.js

```diff
@@ -14,27 +14,53 @@ class Card{
 
 class Deck{
 	constructor(){
+		this.cards = [];
 		//takes in nothing
 		//make storage for your card objects
 	}
 	addCard( suit, faceValue ){
+		var newCard = new Card(suit,faceValue);
+		this.cards.push(newCard);
+		return this.cards.length;
 		//adds a card to the deck
 		//takes in a string suit and faceValue
 		//makes a new card Object from the Card template
 		//adds the card object to the storage in the constructor
 		//returns the amount of cards currently stored
 	}
 	shuffle(){
+		var shuffled = [];
+		for (var arrayIndex=0;arrayIndex<this.cards.length;arrayIndex++){
```

Comment:
wouldn't this only shuffle half the deck?  As arrayIndex is increasing while this.cards.length is decreasing.

it's like 2 cars driving 60 miles between them, at 60 miles per hour towards one another.  they will hit each other in 30 minutes.

really, a while loop would work
``` while( this.cards.length ){ ```

especially since you are not using arrayIndex anywhere.  When you have a loop that you never use the counter anywhere, consider whether the for loop is really the best tool for the job

Reply:
actually, I realize why it didn't do it.  You never actually removed the card from the original array.  You were "slicing", which did nothing but make a copy and do nothing with it.   So what happens is you get a bunch of duplicate entries in your shuffle.

for example, here you have the 6 and 9 of diamonds twice.

<img width="380" alt="screen shot 2019-03-01 at 1 38 06 am" src="https://user-images.githubusercontent.com/10343746/53629584-be112680-3bc2-11e9-886d-534c3a2d6a8a.png">